ethpm_cli/_utils/solc.py

```diff
@@ -89,3 +100,14 @@ def get_contract_types(solc_output: Dict[str, Any]) -> Iterable[str]:
     for source in solc_output:
         for ctype in solc_output[source].keys():
             yield ctype
+
+
+@to_tuple
+def get_contract_types_and_sources(
+    solc_output: Dict[str, Any]
+) -> Iterable[Tuple[str, Any]]:
+    for source in solc_output:
+        for ctype, data in solc_output[source].items():
+            metadata = json.loads(data["metadata"])
+            sources = metadata["sources"].keys()
+            yield (ctype, sources)
```

Comment:
You might consider normalizing the `sources` to be `Path` objects at this stage.  Saves you some `Path` casting elsewhere.